admin-jobs/app/controllers/NewsAlertController.scala

```diff
@@ -1,10 +1,48 @@
 package controllers
 
+import akka.actor.{ActorRef, Props}
+import akka.pattern.ask
+import akka.util.Timeout
+import common.ExecutionContexts
+import model.Cached
+import models.NewsAlertNotification
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.BodyParsers.parse.{json => BodyJson}
 import play.api.mvc._
+import scala.concurrent.duration._
 
-object NewsAlertController extends Controller {
-  def alerts() = Action {
-    Ok("No alert yet")
+trait NewsAlertController extends Controller with ExecutionContexts {
+
+  // Actor is useful here to prevent race condition
+  // when accessing or updating the content of Breaking News
+  // since actor's mailbox acts as a queue
+  val breakingNewsUpdater: ActorRef
+  implicit val actorTimeout = Timeout(30.seconds)
+
+  case class NewsAlertError(error: String)
+  implicit private val ew = Json.writes[NewsAlertError]
+
+  def alerts() = Action.async {
+    (breakingNewsUpdater ? GetAlertsRequest).mapTo[Option[JsValue]].map {
+      case Some(json) => Cached(30)(Ok(json))
+      case None => NoContent
+    }.recover{
+      case _ => InternalServerError(Json.toJson(NewsAlertError("Error while accessing alerts")))
+    }
   }
 
+  def create() : Action[NewsAlertNotification] = Action.async(BodyJson[NewsAlertNotification]) { request =>
+    val receivedNotification : NewsAlertNotification = request.body
+    val result = breakingNewsUpdater ? NewNotificationRequest(receivedNotification)
```

Comment:
I decided here to wait for the answer from the updater actor so the success/failure can be passed by to the endpoint caller. Given the load of this endpoint that shouldn't be an issue.
However if we want to increase the throughput we could fire and forget to the actor and returns an 201(Accepted)